### PR TITLE
[DO NOT MERGE] os/arch/arm/src/imxrt : Periodic Timer Implementation

### PR DESCRIPTION
**Pull request by NitishAmbastha**
Adds PIT support for Chaining timers

To be verified on Master branch
Working on protection

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [5f3cc9327d33212ce85c24aad688d63d8190a1ab] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [5f3cc9327d33212ce85c24aad688d63d8190a1ab] - Code Rule Check Result:
os/arch/arm/src/imxrt/imxrt_pit.c:230: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_pit.c:230: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/arm/src/imxrt/imxrt_pit.c:233: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
os/arch/arm/src/imxrt/imxrt_pit.c:233: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/arm/src/imxrt/imxrt_pit.c:307: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/arm/src/imxrt/imxrt_pit.c:314: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/arm/src/imxrt/imxrt_pit.c:338: ERROR: [SPC_M_KWD] space required before the open brace '{'
os/arch/arm/src/imxrt/imxrt_pit.c:338: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
os/arch/arm/src/imxrt/imxrt_pit.c:345: ERROR: [SPC_M_KWD] space required before the open parenthesis '('

File: os/arch/arm/src/imxrt/imxrt_pit.h [hunk 1]
```diff
@@ -123,7 +138,7 @@ static inline void imxrt_pit_getdefaultconfig(pit_config_t *config)
 	config->enableRunInDebug = false;
 }
 
-#if defined(FSL_FEATURE_PIT_HAS_CHAIN_MODE) && FSL_FEATURE_PIT_HAS_CHAIN_MODE
+//#if defined(FSL_FEATURE_PIT_HAS_CHAIN_MODE) && FSL_FEATURE_PIT_HAS_CHAIN_MODE
```
Comment (sijupunnnoose) [inline os/arch/arm/src/imxrt/imxrt_pit.h]: Why #defined is removed?

**Comment by seinfra** [comment]
Target : [0d330fc358b833ff999e166658928b1bf677f329] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [0d330fc358b833ff999e166658928b1bf677f329] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_pit.c [hunk 2]
```diff
@@ -1,153 +1,479 @@
...
+	.base   = PIT,
+	.irq_id = IMXRT_IRQ_PIT,
+};
+
+bool busy_status[4] = {false, false, false, false,};
+bool ch_status[4] = {false, false, false, false,};
+
+int active = 0;
+
+struct imxrt_pit_lowerhalf_s {
```
Comment (sijupunnnoose) [inline os/arch/arm/src/imxrt/imxrt_pit.c]: Please add proper comments for each structures

File: os/arch/arm/src/imxrt/imxrt_pit.c [hunk 3]
```diff
@@ -1,153 +1,479 @@
...
-
-	/* LTMR64H should be read before LTMR64L */
-	valueH = base->LTMR64H;
-	valueL = base->LTMR64L;
-
-	return (((uint64_t)valueH << 32U) + (uint64_t)(valueL));
-}
-
-#endif /* FSL_FEATURE_PIT_HAS_LIFETIME_TIMER */
+/****************************************************************************
```
Comment (sijupunnnoose) [inline os/arch/arm/src/imxrt/imxrt_pit.c]: File mode correct to 644

File: os/arch/arm/src/imxrt/imxrt_pit.c [hunk 4]
```diff
@@ -1,153 +1,479 @@
...
+
+	/* LTMR64H should be read before LTMR64L */
+	valueH = base->LTMR64H;
+	valueL = base->LTMR64L;
+
+	return (((uint64_t)valueH << 32U) + (uint64_t)(valueL));
+}
+#endif
+
+static int imxrt_pit_settimeout(struct timer_lowerhalf_s *lower, uint32_t timeout)
```
Comment (sijupunnnoose) [inline os/arch/arm/src/imxrt/imxrt_pit.c]: Please add comments for all functions

**Pull request closed without merging**